Update README.md

ghiroxx opened the pull request:
c

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Merge pull request #4 from xes-net/xes-net-patch-2

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Add files via upload

ghiroxx committed:
Add files via upload

ghiroxx committed:
Update Clicca per scaricare vercel 2.json

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

ghiroxx reviewed (commented):
from zipfile import ZipFile
import os

# Define the new frontend React structure
base_path = "/mnt/data/frontend"
os.makedirs(base_path + "/public", exist_ok=True)
os.makedirs(base_path + "/src", exist_ok=True)

# Create base files for a simple React app
files_to_create = {
    base_path + "/package.json": """
{
  "name": "sipnosis-frontend",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build"
  }
}
""",
    base_path + "/src/index.js": """
import React from 'react';
import ReactDOM from 'react-dom/client';
import App from './App';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(<App />);
""",
    base_path + "/src/App.js": """
import React from 'react';

function App() {
  return (
    <div style={{ textAlign: 'center', paddingTop: '50px' }}>
      <h1>Benvenuto su SipnoSis</h1>
      <p>Interfaccia React base funzionante</p>
    </div>
  );
}

export default App;
""",
    base_path + "/public/index.html": """
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>SipnoSis</title>
  </head>
  <body>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root"></div>
  </body>
</html>
"""
}

# Write files
for path, content in files_to_create.items():
    with open(path, "w") as f:
        f.write(content.strip())

# Zip it
zip_path = "/mnt/data/sipnosis-frontend.zip"
with ZipFile(zip_path, 'w') as zipf:
    for root, dirs, files in os.walk(base_path):
        for file in files:
            filepath = os.path.join(root, file)
            arcname = os.path.relpath(filepath, base_path)
            zipf.write(filepath, os.path.join("frontend", arcname))

zip_path

ghiroxx committed:
Add files via upload

ghiroxx committed:
Create react-frontend-template.zip'

import os
from zipfile import ZipFile

# Define folder structure for a basic React frontend project
folder_structure = {
    "frontend": {
        "public": {
            "index.html": """<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    <title>SipnoSis</title>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>"""
        },
        "src": {
            "App.js": """import React from 'react';

function App() {
  return (
    <div>
      <h1>Benvenuto su SipnoSis</h1>
      <p>L’oracolo delle macchie ti aspetta...</p>
    </div>
  );
}

export default App;
""",
            "index.js": """import React from 'react';
import ReactDOM from 'react-dom/client';
import App from './App';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(<React.StrictMode><App /></React.StrictMode>);
"""
        },
        "package.json": """{
  "name": "sipnosis",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  }
}"""
    }
}

# Create folders and files
base_path = "/mnt/data/react-frontend-template"
os.makedirs(base_path, exist_ok=True)

def create_structure(base, structure):
    for name, content in structure.items():
        path = os.path.join(base, name)
        if isinstance(content, dict):
            os.makedirs(path, exist_ok=True)
            create_structure(path, content)
        else:
            with open(path, "w") as f:
                f.write(content)

create_structure(base_path, folder_structure)

# Zip the directory
zip_path = "/mnt/data/react-frontend-template.zip"
with ZipFile(zip_path, 'w') as zipf:
    for root, dirs, files in os.walk(base_path):
        for file in files:
            full_path = os.path.join(root, file)
            rel_path = os.path.relpath(full_path, base_path)
            zipf.write(full_path, arcname=os.path.join("frontend", rel_path))

zip_path

ghiroxx committed:
Update README.md

ghiroxx committed:
Front end

ghiroxx committed:
Update Front end

ghiroxx committed:
Update README.md

ghiroxx reviewed (commented):
from pathlib import Path

# Create the Vercel configuration file content
vercel_config = {
    "version": 2,
    "builds": [
        {
            "src": "index.html",
            "use": "@vercel/static"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "/index.html"
        }
    ]
}

# Define the path to save the file
file_path = Path("/mnt/data/vercel-2.json")

# Save the file
import json
with open(file_path, "w") as f:
    json.dump(vercel_config, f, indent=4)

file_path.name  # Return just the filename for download display

ghiroxx committed:
Update README.md

ghiroxx committed:
Update README.md

xes-net commented on README.md:
```diff
@@ -1 +1,564 @@
-# sipnosis
```
-#sipnosis#-

ghiroxx replied:
> -

xes-net commented on README.md:
```diff
@@ -1 +1,564 @@
-# sipnosis
```
- [ -# spinosi ] -

xes-net replied:
> - [ -# spinosi ] -

ghiroxx commented on Clicca per scaricare vercel 2.json:
```diff
@@ -0,0 +1,31 @@
+      "dest": "frontend/build/index.html"
+    }
+  ]
+}version"
```
[]()

ghiroxx committed:
Update Clicca per scaricare vercel 2.json

ghiroxx committed:
Update Clicca per scaricare vercel 2.json

ghiroxx committed:
Update Clicca per scaricare vercel 2.json

ghiroxx committed:
Update Clicca per scaricare vercel 2.json

ghiroxx committed:
Update Clicca per scaricare vercel.json

ghiroxx committed:
Update Clicca per scaricare vercel.json

ghiroxx committed:
Update README.md

ghiroxx committed:
Update Front end

ghiroxx commented on Scarica README-SIPNOSIS-FINALE.L.md:
```diff
@@ -0,0 +1,108 @@
+
+## 📜 Licenza
+
+Questo progetto è protetto dalla licenza `.L` di Vyra.L – Denominazione di Origine Protetta Mistica™.
```
[](https://github.com/xes-net/sipnosis/pull/2/files)

ghiroxx committed:
Update Front end

ghiroxx committed:
fix: clean html file

ghiroxx committed:
Merge pull request #8 from xes-net/codex/find-and-fix-important-bug

ghiroxx committed:
Add AgorHour single-file FastAPI app

ghiroxx committed:
Merge pull request #11 from xes-net/codex/add-environmental-variable-configuration-guide

ghiroxx committed:
Clean imports and handle OpenAI errors

ghiroxx committed:
chore: add static frontend v2

ghiroxx committed:
Merge pull request #15 from xes-net/codex/update-frontend-to-version-2

vercel[bot] committed:
Implement Vercel Web Analytics integration

## Vercel Web Analytics Integration Implementation

### Overview
Successfully integrated Vercel Web Analytics into the sipnosis project. The site is a plain HTML application without a package manager or build process, so the analytics script was added directly to the HTML file.

### Changes Made

#### Modified Files
- **index.html**: Added Vercel Web Analytics script tag to the `<head>` section

### Implementation Details

**What was done:**
1. Analyzed the project structure and identified it as a plain HTML site (not a Node.js/React project)
2. Added the Vercel Web Analytics script directly to `index.html` using the CDN-hosted script
3. Placed the script in the `<head>` section with `defer` attribute for optimal performance
4. The script automatically configures itself using environment variables set by Vercel

**Why this approach:**
- For plain HTML sites, no package installation is needed
- The script is loaded from Vercel's CDN (`https://cdn.vercel-analytics.com/v1/script.js`)
- Using `defer` attribute ensures the script loads after the page parsing completes, maintaining performance
- No additional code imports or manual initialization needed - the script is auto-configurable

### Technical Notes

- **Target Environment**: Plain HTML site hosted on Vercel
- **Script Source**: `https://cdn.vercel-analytics.com/v1/script.js`
- **Execution**: Client-side only (as required)
- **No Route Support**: The analytics script does not include built-in route tracking (suitable for static HTML)
- **Auto-Configuration**: The script automatically detects the Vercel environment and configures itself

### Verification

- ✅ HTML file is valid and well-formed
- ✅ Script tag is properly placed in the head section
- ✅ No existing functionality was broken
- ✅ Original code structure preserved
- ✅ Script uses defer attribute for optimal loading

### Next Steps

Once deployed to Vercel, the analytics script will:
1. Automatically detect the Vercel environment via environment variables
2. Start collecting web analytics data
3. Make data available in the Vercel Analytics dashboard
4. No additional configuration needed on the frontend

The integration is complete and ready for deployment.

Co-authored-by: Vercel <vercel[bot]@users.noreply.github.com>

ghiroxx committed:
Add GitHub Actions workflow for Python application

This workflow installs Python dependencies, runs linting with flake8, and executes tests using pytest.

ghiroxx committed:
Delete agorhour.py

ghiroxx committed:
Merge pull request #17 from xes-net/codex/add-environmental-variable-configuration-guide

ghiroxx committed:
Update issue templates

ghiroxx committed:
Merge pull request #20 from xes-net/xes-net-patch-2

ghiroxx committed:
Merge pull request #16 from xes-net/vercel/vercel-web-analytics-integrati-8t7wfl

ghiroxx committed:
fix: add stable frontend entrypoint for Vercel

ghiroxx committed:
fix: add static frontend package manifest

ghiroxx committed:
fix: set Vercel static output directory

ghiroxx committed:
docs: replace corrupted README with stable project documentation

ghiroxx committed:
feat: add Flask backend requirements

ghiroxx committed:
feat: add oracle engine module

ghiroxx committed:
feat: add Flask backend API

ghiroxx committed:
chore: add backend environment example

ghiroxx committed:
docs: add backend README

ghiroxx committed:
feat: connect frontend to backend API with local fallback

ghiroxx committed:
feat: add OpenAI SDK dependency for vision oracle

ghiroxx committed:
feat: add AI vision oracle module

ghiroxx committed:
feat: integrate AI vision analysis into oracle API

ghiroxx committed:
feat: render AI vision oracle results in frontend UI

ghiroxx committed:
feat: add Vercel serverless oracle endpoint

ghiroxx committed:
feat: switch frontend to same-origin serverless oracle API

ghiroxx committed:
fix: replace unsafe root index with SipnoSis frontend

ghiroxx committed:
feat: strengthen SipnoSis stain-reading oracle doctrine

ghiroxx committed:
fix: render current oracle API fields in frontend

ghiroxx committed:
feat: add uploaded image preview before oracle reading

ghiroxx committed:
feat: add local ritual history book

ghiroxx committed:
feat: add editable reflection notes to ritual journal

ghiroxx committed:
feat: add Cognitive Bridge MVP clarify interface

ghiroxx committed:
Merge pull request #25 from xes-net/cognitive-bridge-mvp

ghiroxx committed:
feat: add symbol recurrence tracking

ghiroxx committed:
fix: reset pattern panel when history is cleared

ghiroxx committed:
Fix GitHub Actions pytest no-tests failure

ghiroxx committed:
Clean agorhour Python lint style

ghiroxx committed:
Phase 1: extract config module

ghiroxx committed:
Phase 1: extract moderation module

ghiroxx committed:
Phase 1: add moderation tests

ghiroxx committed:
Phase 1: extract AI question generation module

ghiroxx committed:
Phase 1: extract scheduler module

ghiroxx committed:
Phase 1: add AI question tests

ghiroxx committed:
Phase 1: extract lightweight database module

ghiroxx committed:
Phase 1: add database tests

ghiroxx committed:
Phase 1: improve centralized backend app entrypoint

ghiroxx committed:
Phase 1: extract health routes

ghiroxx committed:
Phase 1: extract oracle routes

ghiroxx committed:
Phase 1: register blueprint routes and simplify app entrypoint

ghiroxx committed:
Add protective gitignore for secrets and local files

ghiroxx committed:
Add __init__.py to tests directory for pytest discovery

ghiroxx committed:
Build static bridge routes in Vercel output

ghiroxx committed:
Add Cognitive Bridge landing route

ghiroxx committed:
Add Cognitive Bridge clarify input route

ghiroxx committed:
Add Cognitive Bridge result route

ghiroxx committed:
Document static Vercel structure and Cognitive Bridge route

ghiroxx committed:
Make database module safe without Supabase dependencies

ghiroxx committed:
Make AI question generation safe without OpenAI dependency
